src/render/draw_symbol.js

```diff
@@ -333,7 +334,8 @@ function drawLayerSymbols(painter, sourceCache, layer, coords, isText, translate
             hasHalo
         };
 
-        if (sortFeaturesByKey) {
+        if (hasSortKey && bucket.canOverlap) {
+            sortFeaturesByKey = true;
```

Comment:
I'm just not sure about this particular bit — we need `sortFeaturesByKey` after this loop, so I had to assign it like this.

Reply:
Yeah, this looks good.